pointmatcher/DataPointsFilters/NormalSpace.cpp

```diff
@@ -172,7 +176,12 @@ template <typename T>
 std::size_t NormalSpaceDataPointsFilter<T>::bucketIdx(T theta, T phi) const
 {
 	//Theta = polar angle in [0 ; pi] and Phi = azimuthal angle in [0 ; 2pi]
-	return static_cast<std::size_t>(theta / epsilon) * static_cast<std::size_t>(2. * M_PI / epsilon) + static_cast<std::size_t>(phi / epsilon);
+	assert((theta >= 0.0) && (theta <= static_cast<T>(M_PI)) && (phi >= 0) && (phi <= 2.0*((float) static_cast<T>(M_PI))));
```

Comment:
Sorry, I only noticed it now: you could split the assertion up in multiple assertions, and add output if they fail: https://stackoverflow.com/questions/3692954/add-custom-messages-in-assert

You can still merge it, only if you want to change it.